kolibri/auth/management/commands/importusers.py

```diff
@@ -0,0 +1,141 @@
+        try:
+            # Try lookup by id first, then name
+            return Facility.objects.get(pk=user['facility'])
+        except (Facility.DoesNotExist, ValueError):
+            try:
+                return Facility.objects.get(name=user['facility'])
+            except Facility.DoesNotExist:
+                raise CommandError('Facility matching identifier {facility} was not found'.format(facility=user['facility']))
+    else:
+        return default_facility
+
+def infer_and_create_class(user, facility):
+    if 'class' in user and user['class']:
+        try:
+            # Try lookup by id first, then name
+            classroom = Classroom.objects.get(pk=user['class'], parent=facility)
+        except (Classroom.DoesNotExist, ValueError):
+            try:
+                classroom = Classroom.objects.get(name=user['class'], parent=facility)
+            except Classroom.DoesNotExist:
+                classroom = Classroom.objects.create(name=user['class'], parent=facility)
+        return classroom
+
+def create_user(i, user, default_facility=None):
+    validate_username(user)
+
+    if i == 0 and all(key == val or val is None for key, val in user.items()):
+        # Check whether the first row is a header row or not
+        # Either each key will be equal to the value
+        # Or the header is not included in the CSV, so it is None
+        return False
+
+    facility = infer_facility(user, default_facility)
+    classroom = infer_and_create_class(user, facility)
+    try:
+        username = user['username']
+        user_obj = FacilityUser.objects.get(username=username, facility=facility)
+        logging.warn('Tried to create a user with the username {username} in facility {facility}, but one already exists'.format(
+            username=username,
+            facility=facility
+        ))
+        if classroom:
+            classroom.add_member(user_obj)
+        return False
+    except FacilityUser.DoesNotExist:
+        new_user = FacilityUser.objects.create(
+            full_name=user.get('full_name', ''),
+            username=username,
+            facility=facility,
+        )
+        password = user.get('password', DEFAULT_PASSWORD)
+        new_user.set_password(password)
+        new_user.save()
+        if classroom:
+            classroom.add_member(new_user)
+        logging.info('User created with username {username} in facility {facility} with password {password}'.format(
+            username=username,
+            facility=facility,
+            password=password,
+        ))
+        return True
+
+class Command(BaseCommand):
+    help = """
+    Imports a user list from CSV file and creates
+    them on a specified or the default facility.
+
+    Requires CSV file data in this form:
+    <full_name>,<username>,<password>,<facility>,<class>
+
+    Less information can be passed if the headers are specified:
+    full_name,username,password,class
+    <full_name>,<username>,<password>,<class>
```

Comment:
I don't understand this "Less information can be passed" ? :)

Reply:
Fewer columns.